packages/cli/src/cmds/validator/keymanager/decryptKeystoreDefinitions/index.ts

```diff
@@ -0,0 +1,105 @@
+import {spawn, ModuleThread, Pool, QueuedTask, Worker} from "@chainsafe/threads";
+import {SignerLocal, SignerType} from "@lodestar/validator";
+import bls from "@chainsafe/bls";
+import {LocalKeystoreDefinition} from "../interface.js";
+import {clearKeystoreCache, loadKeystoreCache, writeKeystoreCache} from "../keystoreCache.js";
+import {lockFilepath, unlockFilepath} from "../../../../util/lockfile.js";
+import {defaultPoolSize} from "./poolSize.js";
+import {DecryptKeystoreWorkerAPI, KeystoreDecryptOptions} from "./types.js";
+
+/**
+ * Decrypt keystore definitions using a threadpool
+ */
+export async function decryptKeystoreDefinitions(
+  keystoreDefinitions: LocalKeystoreDefinition[],
+  opts: KeystoreDecryptOptions
+): Promise<SignerLocal[]> {
+  if (opts.cacheFilePath) {
+    try {
+      const signers = await loadKeystoreCache(opts.cacheFilePath, keystoreDefinitions);
+      if (opts?.onDecrypt) {
+        opts?.onDecrypt(signers.length - 1);
+      }
+      opts.logger.debug("Loaded keystores via keystore cache");
+      return signers;
+    } catch {
+      // Some error loading the cache, ignore and invalidate cache
+      await clearKeystoreCache(opts.cacheFilePath);
+    }
+  }
+
+  const signers = new Array(keystoreDefinitions.length) as SignerLocal[];
+  const passwords = new Array(keystoreDefinitions.length) as string[];
+  const tasks: QueuedTask<ModuleThread<DecryptKeystoreWorkerAPI>, Uint8Array>[] = [];
+  const errors: Error[] = [];
+  const pool = Pool(
+    () =>
+      spawn<DecryptKeystoreWorkerAPI>(new Worker("./worker.js"), {
+        // A Lodestar Node may do very expensive task at start blocking the event loop and causing
+        // the initialization to timeout. The number below is big enough to almost disable the timeout
+        timeout: 5 * 60 * 1000,
+      }),
+    defaultPoolSize
+  );
+  for (const [index, definition] of keystoreDefinitions.entries()) {
+    try {
+      lockFilepath(definition.keystorePath);
+    } catch (e) {
+      if (opts.ignoreLockFile) {
+        opts.logger.warn("Keystore forcefully loaded even though lockfile could not be acquired", {
```

Comment:
Is this the right warning message? shouldn't it rather be

```suggestion
        opts.logger.warn("Keystore forcefully loaded even though lockfile exists", {
```

Mostly, not sure what "could not be acquired" is trying to say